examples/01-basic/04-saving-loading/App.tsx

```diff
@@ -0,0 +1,53 @@
+import { PartialBlock } from "@blocknote/core";
+import { BlockNoteView, createBlockNoteEditor } from "@blocknote/react";
+import "@blocknote/react/style.css";
+import { useEffect, useMemo, useState } from "react";
+
+async function saveToStorage(jsonBlocks: any[]) {
+  // Save contents to local storage. You might want to debounce this or replace with a call to your API / database
+  localStorage.setItem("editorContent", JSON.stringify(jsonBlocks));
+}
+
+async function loadFromStorage() {
+  // Gets the previously stored editor contents
+  const storageString = localStorage.getItem("editorContent");
+  return storageString
+    ? (JSON.parse(storageString) as PartialBlock[])
+    : undefined;
+}
+
+export default function App() {
+  const [initialContent, setInitialContent] = useState<
+    PartialBlock[] | undefined | "loading"
+  >("loading");
+
+  // Loads the previously stored editor contents
+  useEffect(() => {
+    loadFromStorage().then((content) => {
+      setInitialContent(content);
+    });
+  }, []);
+
+  // Creates a new editor instance.
+  // We use useMemo + createBlockNoteEditor instead of useBlockNote so we can delay the creation of the editor until the initial content is loaded.
+  const editor = useMemo(() => {
+    if (initialContent === "loading") {
+      return undefined;
+    }
+    return createBlockNoteEditor({ initialContent });
```

Comment:
This seems like a bit of a weird pattern - wouldn't it make more sense for you to be able to e.g. pass editor options in an async function? Then BlockNote can wait to init the editor instead of the consumer having to do it.

Reply:
@YousefED tbd